music_this_week_app/views.py

```diff
@@ -79,23 +80,35 @@ def callback(request):
 
 def setup(request):
     """Displays search args and a search button, which links to /search"""
-
+    #get user ip address to find user location 
+    x_forwarded_for = request.META.get('HTTP_X_FORWARDED_FOR')
+    if x_forwarded_for:
+        ip = x_forwarded_for.split(',')[0]
+    else:
+        ip = request.META.get('REMOTE_ADDR')
+    #handle invalid IP or ip == 127.0.0.1 because running locally 
+    try:
+        g = GeoIP2()
+        location = g.city(ip)
+    except AddressNotFoundError:
+        location = {'city' : "San Francisco"}
     # Fetch user data
     pc = request.session.get('pc')
     if pc is None:
         print("ERROR: setup called without pc session. Redirecting home.")
         return HttpResponseRedirect('/')
 
-    context = pc.user_info
+    context = dict(pc.user_info, **{'location' : location['city']}) 
     return render(request, 'music_this_week/setup.html', context)
 
 def search(request):
     """ Search for shows and create playlist then redirect to it"""
     # Parse request for search arguments
     search_args = dict(request.GET.items())
-
+    print(search_args)
```

Comment:
filtering occurs in eventFinder.py.

HTML will only submit the key/val
  ```u'latlng_checkbox': u'on', ```
if the checkbox is ticked (which only happens if we have the users location permissions)
else no key ```u'latlng_checkbox': u'on', ``` will be in the dict

seems suboptimal, but I'm not sure how we can get around it (unless I am missing something). Anyway, also handled in eventFinder.py